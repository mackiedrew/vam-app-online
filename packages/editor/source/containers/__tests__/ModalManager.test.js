import {
  ModalManager as Subject,
  mapStateToProps,
  mapDispatchToProps
} from "../ModalManager";
jest.mock("../../components/RemoveTrackModal", () => "RemoveTrackModal");

const baseMockProps = {
  clearModal: jest.fn(),
  removeTrack: jest.fn(),
  data: { trackId: "123ABC" }
};

const noneMock = {
  ...baseMockProps,
  type: "NONE"
};

const removeTrackMock = {
  ...baseMockProps,
  type: "REMOVE_TRACK"
};

describe("<ModalManager />", () => {
  describe("renders without crashing", () => {
    it("as NONE modal", () => {
      shallow(<Subject {...noneMock} />);
    });
    it("as REMOVE_TRACK modal", () => {
      shallow(<Subject {...removeTrackMock} />);
    });
  });

  describe("combined modal removal functions fire correctly", () => {
    it("removeTrackAndModal", () => {
      const mockRemoveTrack = sinon.spy();
      const mockClearModal = sinon.spy();
      const subject = shallow(
        <Subject
          {...removeTrackMock}
          clearModal={mockRemoveTrack}
          removeTrack={mockClearModal}
        />
      );
      subject.instance().removeTrackAndModal();
      expect(mockRemoveTrack.called).toBe(true);
      expect(mockClearModal.called).toBe(true);
    });

    it("removeTrackAndModal without trackId", () => {
      const mockRemoveTrack = sinon.spy();
      const mockClearModal = sinon.spy();
      const subject = shallow(
        <Subject
          {...removeTrackMock}
          clearModal={mockRemoveTrack}
          data={{}}
          removeTrack={mockClearModal}
        />
      );
      subject.instance().removeTrackAndModal();
      expect(mockRemoveTrack.called).toBe(true);
      expect(mockClearModal.called).toBe(true);
    });
  });

  describe("renders correctly", () => {
    it("as NONE modal", () => {
      const tree = renderer.create(<Subject {...noneMock} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
    it("as REMOVE_TRACK modal", () => {
      const tree = renderer.create(<Subject {...removeTrackMock} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe("mapStateToProps()", () => {
    const mockState = {
      ui: {
        modalType: 1,
        modalData: 2
      }
    };

    it("returns object with expected properties", () => {
      const { type, data } = mapStateToProps(mockState);
      expect(type).toBe(1);
      expect(data).toBe(2);
    });
  });

  describe("mapDispatchToProps()", () => {
    it("returns object with expected properties", () => {
      const mockDispatch = sinon.spy();
      const { clearModal, removeTrack } = mapDispatchToProps(mockDispatch);
      expect(typeof clearModal).toBe("function");
      expect(typeof removeTrack).toBe("function");
    });
  });
});
