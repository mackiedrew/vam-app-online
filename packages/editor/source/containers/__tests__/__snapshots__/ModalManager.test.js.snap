// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ModalManager /> renders correctly as NONE modal 1`] = `
<div
  className="modal-manager"
>
  
</div>
`;

exports[`<ModalManager /> renders correctly as REMOVE_TRACK modal 1`] = `
<div
  className="modal-manager"
>
  <RemoveTrackModal
    clearModal={[Function]}
    fileName="this track"
    removeTrack={[Function]}
  />
</div>
`;
